model/textparse/protobufparse_test.go

```diff
@@ -154,6 +154,78 @@ metric: <
   timestamp_ms: 1234568
 >
 
+`,
+
+		`name: "test_float_histogram"
+help: "Test float histogram with many buckets removed to keep it manageable in size."
+type: HISTOGRAM
+metric: <
+  histogram: <
+    sample_count: 175
+	sample_count_float: 175.0
+    sample_sum: 0.0008280461746287094
+    bucket: <
+      cumulative_count_float: 2.0
+      upper_bound: -0.0004899999999999998
+    >
+    bucket: <
+      cumulative_count_float: 4.0
+      upper_bound: -0.0003899999999999998
+      exemplar: <
+        label: <
+          name: "dummyID"
+          value: "59727"
+        >
+        value: -0.00039
+        timestamp: <
+          seconds: 1625851155
+          nanos: 146848499
+        >
+      >
+    >
+    bucket: <
+      cumulative_count_float: 16
+      upper_bound: -0.0002899999999999998
+      exemplar: <
+        label: <
+          name: "dummyID"
+          value: "5617"
+        >
+        value: -0.00029
+      >
+    >
+    schema: 3
+    zero_threshold: 2.938735877055719e-39
+    zero_count_float: 2.0
+    negative_span: <
+      offset: -162
+      length: 1
+    >
+    negative_span: <
+      offset: 23
+      length: 4
+    >
+    negative_delta: 1
+    negative_delta: 3
+    negative_delta: -2
+    negative_delta: -1
```

Comment:
Why do we have deltas here and buckets above in the same histograms? I think only one of they should be set

Reply:
It's possible to present both conventional `Buckets` and new deltas. A consumer not capable of parsing the new native histograms will then simple parse the conventional histogram, while newer consumers can choose to use the native histogram rather than the conventional buckets.

Reply:
Using the conventional buckets is (for the time being) even required if you want to use exemplars, as is tested in this case.

Reply:
Of course, this test example is inconsistent because the conventional buckets are for a float histogram, while the native histogram uses integer deltas (but should use float counts). I guess this will become clearer once the comments above are addressed.